suno-factory/backend/src/services/parser.py

```
@@ -0,0 +1,55 @@
+                return FileParser._extract_pdf(content)
+            if filename.endswith(".docx"):
+                return FileParser._extract_docx(content)
+            if filename.endswith((".txt", ".md")):
+                return content.decode("utf-8")
+            raise HTTPException(
+                status_code=400,
+                detail=f"Unsupported file type: {filename.split('.')[-1]}",
+            )
+        except HTTPException:
+            raise
+        except Exception as exc:
```

Ruff (0.15.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)